<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Simple Promise</title>
</head>

<body>

  <script>
    function divide(a, b) {
      return new Promise((resolve, reject) => {
        if (b === 0) {
          reject(new Error('Cannot divide by 0!'))
        }

        const result = a / b;
        resolve(result);
      });
    }

    console.log('start');
    divide(8, 2)
      .then((result) => {
        console.log('async result', result); // note: this log is printed after `end` due to the async behavior
      })
      .catch((error) => {
        console.error('error handler', error);
      });
    console.log('end');

  </script>
</body>

</html>
